### PR TITLE
Miscellaneous refactoring

### PR DESCRIPTION
**Pull request by bitfriend**
- Replace `String::as_str()` with string reference
- Replace `debugPrint()` with `logging` package

### CODE REVIEW

File: app/lib/features/onboarding/pages/register_page.dart [hunk 1]
```diff
@@ -240,7 +240,7 @@ class _RegisterPageState extends ConsumerState<RegisterPage> {
                                   TextSpan(
                                     recognizer: TapGestureRecognizer()
                                       ..onTap = () {
-                                        debugPrint('Terms of Service"');
+                                        _log.info('Terms of Service"');
```
Comment (gnunicorn) [inline app/lib/features/onboarding/pages/register_page.dart:243]: these aren't really meant to be logged but instead replaced with actual content ... the print is there for us to notice ... but whatever.

File: app/packages/rust_sdk/lib/acter_flutter_sdk.dart [hunk 2]
```diff
@@ -372,20 +372,20 @@ class ActerSdk {
       // android api 30 is working here
       return DynamicLibrary.open(libName);
     } catch (e1) {
-      debugPrint('DynamicLibrary.open by lib name failed: $e1');
+      _log.severe('DynamicLibrary.open by lib name failed: $e1');
```
Comment (gnunicorn) [inline app/packages/rust_sdk/lib/acter_flutter_sdk.dart:375]: nope. the idea of `_log` was to go _through_ the API, but these things here are failing so we can't log through `_log`. These need to work regardless.

```suggestion
      debugPrint('DynamicLibrary.open by lib name failed: $e1');
```

File: app/packages/rust_sdk/lib/acter_flutter_sdk.dart [hunk 3]
```diff
@@ -372,20 +372,20 @@ class ActerSdk {
...
       try {
         // android api 23 is working here
         final String nativeLibDir = await _getNativeLibraryDirectory();
         return DynamicLibrary.open('$nativeLibDir/$libName');
       } catch (e2) {
-        debugPrint('DynamicLibrary.open from /data/app failed: $e2');
+        _log.severe('DynamicLibrary.open from /data/app failed: $e2');
```
Comment (gnunicorn) [inline app/packages/rust_sdk/lib/acter_flutter_sdk.dart:381]:
```suggestion
        debugPrint('DynamicLibrary.open from /data/app failed: $e2');
```

File: app/packages/rust_sdk/lib/acter_flutter_sdk.dart [hunk 4]
```diff
@@ -372,20 +372,20 @@ class ActerSdk {
...
         try {
           // android api 8 (2010) is working here
           final PackageInfo pkgInfo = await PackageInfo.fromPlatform();
           final String pkgName = pkgInfo.packageName;
           return DynamicLibrary.open('/data/data/$pkgName/$libName');
         } catch (e3) {
-          debugPrint('DynamicLibrary.open from /data/data failed: $e3');
+          _log.severe('DynamicLibrary.open from /data/data failed: $e3');
```
Comment (gnunicorn) [inline app/packages/rust_sdk/lib/acter_flutter_sdk.dart:388]:
```suggestion
          debugPrint('DynamicLibrary.open from /data/data failed: $e3');
```

**Review by gnunicorn — APPROVED** [review]
some things you were too eager and didn't notice the context it is in. these need to be reverted.

File: app/lib/features/onboarding/pages/register_page.dart (hunk 1 above)
Comment (bitfriend) [reply app/lib/features/onboarding/pages/register_page.dart]: Fixed by https://github.com/acterglobal/a3/pull/1495/commits/d67ca486cf571f3de5170bc80b20bbbb5409502c

File: app/packages/rust_sdk/lib/acter_flutter_sdk.dart (hunk 2 above)
Comment (bitfriend) [reply app/packages/rust_sdk/lib/acter_flutter_sdk.dart]: Fixed by https://github.com/acterglobal/a3/pull/1495/commits/a6c2799abeb8622694e4e2c9c9ef14dd9cea0007

File: app/packages/rust_sdk/lib/acter_flutter_sdk.dart (hunk 3 above)
Comment (bitfriend) [reply app/packages/rust_sdk/lib/acter_flutter_sdk.dart]: Fixed by https://github.com/acterglobal/a3/pull/1495/commits/a6c2799abeb8622694e4e2c9c9ef14dd9cea0007

File: app/packages/rust_sdk/lib/acter_flutter_sdk.dart (hunk 4 above)
Comment (bitfriend) [reply app/packages/rust_sdk/lib/acter_flutter_sdk.dart]: Fixed by https://github.com/acterglobal/a3/pull/1495/commits/a6c2799abeb8622694e4e2c9c9ef14dd9cea0007

**Pull request merged**